src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -8559,84 +8719,39 @@ protected MethodGroupResolution ResolveExtension(
                 {
-                    // Create a set of arguments for overload resolution of the
-                    // extension methods that includes the "this" parameter.
-                    actualArguments = AnalyzedArguments.GetInstance();
-                    CombineExtensionMethodArguments(left, analyzedArguments, actualArguments);
+                    // Create a set of arguments for overload resolution including the receiver.
+                    actualMethodArguments = AnalyzedArguments.GetInstance();
+                    CombineExtensionMethodArguments(left, analyzedArguments, actualMethodArguments);
                 }
 
-                var extensionMethodResult = resolveOverloads(this, methodGroup, actualArguments, options, returnType, returnRefKind, in callingConvention, expression, diagnostics);
-                diagnostics.Free();
-
-                // If the search in the current scope resulted in any applicable method (regardless of whether a best
-                // applicable method could be determined) then our search is complete. Otherwise, store aside the
-                // first non-applicable result and continue searching for an applicable result.
-                if (extensionMethodResult.HasAnyApplicableMethod)
-                {
-                    firstResult.Free();
-                    actualArguments?.Free();
-                    lookupResult.Free();
-                    classicExtensionLookupResult.Free();
-                    return extensionMethodResult;
-                }
-                else if (firstResult.IsEmpty)
-                {
-                    firstResult = extensionMethodResult;
-                }
-                else
-                {
-                    // Neither the first result, nor applicable. No need to save result.
-                    extensionMethodResult.Free();
-                }
-            }
-
-            Debug.Assert((actualArguments == null) || !firstResult.IsEmpty);
-            lookupResult.Free();
-            classicExtensionLookupResult.Free();
-
-            return firstResult;
-
-            static MethodGroupResolution resolveOverloads(Binder binder, MethodGroup methodGroup, AnalyzedArguments actualArguments, OverloadResolution.Options options,
-                TypeSymbol? returnType, RefKind returnRefKind, in CallingConventionInfo callingConvention, SyntaxNode expression, BindingDiagnosticBag diagnostics)
-            {
                 var overloadResolutionResult = OverloadResolutionResult<MethodSymbol>.GetInstance();
-                if (binder.AllowRefOmittedArguments(methodGroup.Receiver))
-                {
-                    options |= OverloadResolution.Options.AllowRefOmittedArguments;
-                }
-
                 CompoundUseSiteInfo<AssemblySymbol> overloadResolutionUseSiteInfo = binder.GetNewCompoundUseSiteInfo(diagnostics);
                 binder.OverloadResolution.MethodInvocationOverloadResolution(
```

Comment:
>MethodInvocationOverloadResolution [](http://example.com/codeflow?start=42&length=34)

We should test scenarios when receiver parameter is the only parameter and it has `params` modifier (can happen in metadata).